application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -322,6 +335,34 @@ private void updateNestedRuleChains() {
         }
     }
 
+    private void updateCheckRelationNode() {
+        PageDataIterable<RuleNode> ruleNodesIterator = new PageDataIterable<>(
+                link -> ruleChainService.findAllRuleNodesByType(TbCheckRelationNode.class.getName(), link), 1024);
+        ruleNodesIterator.forEach(ruleNode -> {
+            TbCheckRelationNodeConfiguration configNode = JacksonUtil.convertValue(ruleNode.getConfiguration(), TbCheckRelationNodeConfiguration.class);
+            RuleChain targetRuleChain = ruleChainService.findRuleChainById(TenantId.SYS_TENANT_ID, ruleNode.getRuleChainId());
+            if (targetRuleChain != null) {
+                TenantId tenantId = targetRuleChain.getTenantId();
+                String directionNode = configNode.getDirection();
+                if (configNode.isCheckForSingleEntity()) {
+                    configNode.setDirection(directionNode.equals(EntitySearchDirection.FROM.name())
```

Comment:
switch directionNode with EntitySearchDirection.FROM.name(), to validate null values